### PR TITLE
Address inconsistent behavior of generated resources files

### PR DESCRIPTION
**Pull request by azymnis**
This deals with issue #296

### CODE REVIEW

**Comment by johnynek** [comment]
test this please

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

**Comment by johnynek** [comment]
👍 

Will merge when green.

File: test/src/main/resources/scala/test/BUILD [hunk 1]
```diff
@@ -1 +1,8 @@
 exports_files(["byes", "hellos", "hellos-and-byes.jar", "more-byes", "more-hellos"])
+
+genrule(
```
Comment (johnynek) [inline test/src/main/resources/scala/test/BUILD]: getting fancy: generated resources...

Comment (azymnis) [reply test/src/main/resources/scala/test/BUILD]: Yeah admittedly this is a subtle corner case... We somehow still managed to hit it though

**Comment by azymnis** [comment]
@ittaiz @johnynek yo guys I don't want to sound like the grouch here but 2 hour travis builds are not optimal. What can we do to speed these up?

**Comment by johnynek** [comment]
I restarted it. Yeah, the build is really slow. I tried to speed it up using workers. The clean is pretty expensive, also the bazel install is expensive.

Happy to take any PRs on this.

**Comment by ittaiz** [comment]
Also,
Resolving issue #76 will probably help some of this thanks to Bazel caching
